otx/algorithms/classification/adapters/dor/tasks/inference.py

```diff
@@ -0,0 +1,423 @@
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import io
+import logging
+import math
+import os
+import shutil
+import tempfile
+from typing import Dict, Optional
+
+import torch
+import torchreid
+from scripts.default_config import (
+    get_default_config,
+    imagedata_kwargs,
+    merge_from_files_with_base,
+    model_kwargs,
+)
+from torchreid.apis.export import export_ir, export_onnx
+from torchreid.metrics.classification import score_extraction
+from torchreid.utils import load_pretrained_weights
+
+from otx.algorithms.classification.adapters.dor.utils.monitors import (
+    DefaultMetricsMonitor,
+    StopCallback,
+)
+from otx.algorithms.classification.adapters.dor.utils.parameters import (
+    DORClassificationParameters,
+)
+from otx.algorithms.classification.adapters.dor.utils.utils import (
+    DORClassificationDataset,
+    InferenceProgressCallback,
+    active_score_from_probs,
+    force_fp32,
+    get_hierarchical_predictions,
+    get_multiclass_predictions,
+    get_multihead_class_info,
+    get_multilabel_predictions,
+    sigmoid_numpy,
+    softmax_numpy,
+)
+from otx.api.configuration import cfg_helper
+from otx.api.configuration.helper.utils import ids_to_strings
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.inference_parameters import InferenceParameters
+from otx.api.entities.metadata import FloatMetadata, FloatType
+from otx.api.entities.model import (
+    ModelEntity,
+    ModelFormat,
+    ModelOptimizationType,
+    ModelPrecision,
+    OptimizationMethod,
+)
+from otx.api.entities.result_media import ResultMediaEntity
+from otx.api.entities.resultset import ResultSetEntity
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.task_environment import TaskEnvironment
+from otx.api.entities.tensor import TensorEntity
+from otx.api.entities.train_parameters import default_progress_callback
+from otx.api.serialization.label_mapper import label_schema_to_bytes
+from otx.api.usecases.evaluation.metrics_helper import MetricsHelper
+from otx.api.usecases.tasks.interfaces.evaluate_interface import IEvaluationTask
+from otx.api.usecases.tasks.interfaces.export_interface import ExportType, IExportTask
+from otx.api.usecases.tasks.interfaces.inference_interface import IInferenceTask
+from otx.api.usecases.tasks.interfaces.unload_interface import IUnload
+from otx.api.utils.argument_checks import (
+    DatasetParamTypeCheck,
+    check_input_parameters_type,
+)
+from otx.api.utils.labels_utils import get_empty_label
+from otx.api.utils.vis_utils import get_actmap
+
+logger = logging.getLogger(__name__)
+
+
+class DORClassificationInferenceTask(IInferenceTask, IEvaluationTask, IExportTask, IUnload):
```

Comment:
I know this is not used externally and exists only for NNCFTask. Can't you just move this to nncf.py without having to put it in a separate file? If the file exists separately, it appears to be used elsewhere. (It's just a recommendation..?)

Reply:
nncf.py would be so long. I want to rename it just as nncf_inference.py. What do you think?

Reply:
It seems good to combine them into one file or the class (The detection went like this.) itself.

Reply:
Could you look at this again?